tasks/Task.hpp

```diff
@@ -108,8 +114,47 @@ namespace usbl_evologics {
          * before calling start() again.
          */
         void cleanupHook();
-        void storePermanently();
-        void writeOutPosition();
+
+        /** Process I/O data
+         *
+         */
+        void processIO();
+
+        /** Get communication parameters
+         *
+         *  @return AcousticChannel with performance.
+         */
+        AcousticChannel getAcousticChannelparameters(void);
+
+        /** Update parameters on device.
+         *
+         * Compare actual with desired settings before update.
+         * @param desired_setting, parameters that should be applied on device.
+         * @param actual_setting, parameters present in device that will be used for compare.
+         */
+        void updateDeviceParameters(DeviceSettings const &desired_setting, DeviceSettings const &actual_setting);
+
+        /** Verify if free transmission buffer is big enough to support message.
```

Comment:
The usbl has an internal buffer for store data till it can be transmitted over the acoustic link. The free transmission buffer and the overflow counter are outputted and can be monitored. Other storage buffer can be implemented in the component to avoid the overflow of device's buffer, but IMO the point to deal is the flow rate control of data.
IMO, it should be done by the component that want send data to remote device, by monitoring the free transmission buffer. 
What do you think?